qpicard/src/org/qcmg/picard/SAMFileReaderFactory.java

```diff
@@ -27,7 +27,7 @@
 import org.qcmg.common.log.QLogger;
 import org.qcmg.common.log.QLoggerFactory;
 import org.qcmg.common.string.StringUtils;
-import org.qcmg.protocol.s3.S3AwareURLStreamHandlerFactory;
+import org.qcmg.picard.s3.S3AwareURLStreamHandlerFactory;
```

Comment:
I really don't understand this move. The `s3` package is for low-level AWS operations - for example, all it ever throws is IOException. `picard` is specifically for handling fastq/bam/sam sequence data which is a way more specialized task. By moving this out of `qio` into `qpicard` that means any time I want to access ANY s3 file stream, whether it contains a VCF, text or whatever, I have to import a sequence data handling package. That just seems wrong to me.
A lot of my reluctance I think comes from my (very imperfect) understanding of what problem this is attempting to solve. It seems like if the issue is multiple paths to the same dependency, that is something that should be fixed in the build configuration (with excludes) rather than restructuring packages themselves.